.github/workflows/haskell.yml

```diff
@@ -158,7 +158,9 @@ jobs:
         echo "$HOME/.local/bin" >> $GITHUB_PATH
 
     - name: Run fourmolu
-      run: ./scripts/fourmolize.sh
+      run: |
+        git fetch --no-tags --prune --depth=1 origin +refs/heads/master:refs/remotes/origin/master
```

Comment:
Why this fetch? I imagine it's gonna have some impact on the `git diff` ran below as part of `formolize.sh`, but how will it make a difference?

Reply:
It is needed for the GitHub action. By default, the `checkout@v4` action only fetches a single commit from the feature branch, see: https://github.com/actions/checkout. However, since we want to run `fourmolu` on changed files only compared to `origin/master`, we need to have access to `origin/master` and this is how we achieve that. Otherwise, the action would fail and complain with `bad revision: origin/master`.